### ISSUE TITLE
No longer maintained?

### ISSUE THREAD

**Issue by BadgerOps** [churchers/vm-bhyve#544]
Hello @churchers! 

I am a huge fan of vm-bhyve, and greatly appreciate the work you've done to build and maintain this.

From what I can see, it appears that there hasn't been any updates or maintenance of this repo for a couple of years.

I am happy to help contribute my time to reviewing and testing PR's and issues if that would be helpful. I'll shoot you an email to your posted address as well as this gh issue.

Again, thank you for this tool, it is quite helpful!

-BadgerOps

**Comment by BadgerOps** [comment]
I was looking at the freshports page, and it looksl ike the recent commits are actually in the ports tree, not specifically in this repo - but I could be misreading them.

**Comment by tommarcoen** [comment]
Perhaps @driesmp or @bapt could provide some information on this?

**Comment by antranigv** [comment]
Well, to be fair, vm-bhyve is kind of *complete*, so there's not much work to do, other than fixing bugs.

**Comment by driesmp** [comment]
vm-bhyve-devel is just tracking the latest commit to the repo, while the normal version, vm-bhyve is tracking the releases.

**Comment by BadgerOps** [comment]
Hi @antranigv I guess I was going off the PR's stacking up without any responses. Core being feature complete is great, but it doesn't seem like anyone is actively maintaining which is what prompted my question!

Thanks,

-BadgerOps

**Comment by metalefty** [comment]
@churchers @mateuszkwiatkowski I'd love to volunteer to maintain vm-bhyve. I'm recognized as meta@ in FreeBSD community and a ports committer. I'm also a core developer of xrdp and an employed AlmaLinux developer. If you guys are comfortable with that, could you give me some privileges? I would like to merge some obviously useful PRs such as #580, #553, etc.

**Comment by mateuszkwiatkowski** [comment]
@metalefty Thank you for engagement in vm-bhyve community. While I'd love to have someone to help with testing and integrating PRs I don't have GH privileges required to add you as maintainer. We need to wait for @churchers.

**Comment by driesmp** [comment]
Same for me, I tried reaching out to churchers a while back as well to propose this in helping to close issues / PR's. 
Would be very helpful for @metalefty and/or me to have some additional privileges on the repo :).

**Comment by bapt** [comment]
if needed we can move this repo under the FreeBSD org on github and provide the right privileges to anyone who needs, (I mean if @churchers is not working on it anymore)

**Comment by mateuszkwiatkowski** [comment]
I believe it would be reasonable to fork vm-bhyve under the FreeBSD organization and continue its development there, considering its widespread use within the FreeBSD community. I think @churchers  last activity was in 2022: https://github.com/churchers/vm-bhyve/commit/bda366b92af0cce59f3981da50b96b4d0e8e2ad4

**Comment by antranigv** [comment]
Please keep in mind that the beauty of vm-bhyve is that it… doesn’t change much. I don’t think it needs any new features for 99% of the use cases. And for that last 1%, well, bhyve_options does exist. I’m not saying the FreeBSD project shouldn’t fork, I’m just saying we should have a really strong reason to do that. Just my 2 cents. And I’m sure you folks will do the right decision. Sent from my iPhoneOn 5 Mar 2025, at 8:16 PM, Mateusz Kwiatkowski ***@***.***> wrote:﻿
  I believe it would be reasonable to fork vm-bhyve under the FreeBSD organization and continue its development there, considering its widespread use within the FreeBSD community. I think @churchers  last activity was in 2022: bda366b—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>



mateuszkwiatkowski left a comment (churchers/vm-bhyve#544)
I believe it would be reasonable to fork vm-bhyve under the FreeBSD organization and continue its development there, considering its widespread use within the FreeBSD community. I think @churchers  last activity was in 2022: bda366b

**Comment by metalefty** [comment]
I don't think so strong reasons will be required because this would be a friendly fork rather than a hostile fork due to community fragmentation. If anything happens to @mateuszkwiatkowski (collaborator), nobody else can merge pull requests, close issues, or perform other activities that require write access. IMO, this is reason enough to fork.

I don’t think many new features or developments will happen, even if this repository is forked within the FreeBSD community. I believe every developer here understands that it shouldn't undergo significant changes, but rather that the focus should be on fixing bugs, housekeeping, and adding only a few new features while keeping compatibility.

To be clear, I’m not personally in favor of forking at the moment. I believe it would be better if @churchers returned and, with his agreement, chose to manage this repository within the @FreeBSD organization of his own volition. I would suggest that we try our best to reach out to him first, and then consider forking. Has anyone tried reaching out already?

**Comment by metalefty** [comment]
I think it might be better to set a deadline if we make a friendly fork under the @freebsd organization.  We're already waiting for @churchers back at least 1 years and his last activity was 2 years ago. I already tried to reach out to him 2 months ago. 

If and when he comes back, of course, it will be up to him to decide what to do (back to his repo and ask him to add some people as co-admin, or add him as admin to the fork).

@driesmp @bapt @mateuszkwiatkowski 
What do you think? How long much longer should we wait him for if we do fork? 2 weeks? 1 month? 2months or more?

**Comment by bapt** [comment]
done: https://github.com/freebsd/vm-bhyve

**Comment by mateuszkwiatkowski** [comment]
I asked @mzary to update ports to reflect the new location. Thanks, everyone, for the conversation!

**Comment by driesmp** [comment]
Thanks @bapt, I guess there is no way to keep issue history and PR's? (like the ones that are still open in the current repo)

**Comment by bapt** [comment]
no it is impossible, the only way to do it would have been for @churchers  to give me the full ownership on the repos and allow me to migrate., as for the wiki, I think there is a way but I am not sure yet

**Comment by bapt** [comment]
wiki migrated

**Comment by mzary** [comment]
> I asked [@mzary](https://github.com/mzary) to update ports to reflect the new location. Thanks, everyone, for the conversation!

Mateusz,  thank you for the invitation to file a PR via Bugzilla with the port update. Unfortunately, I am not the port's maintainer, nor do I have the commit bit, so all updates need to be approved by @driesmp first and are likely committed by him as well later. To streamline the process, I'll leave the repository rebase to @driesmp.

In the past, I submitted a few port updates because there hadn't been any releases for quite some time, and there was a general impression that the project was dead. Now that @mateuszkwiatkowski is regularly preparing new releases, that concern no longer applies.

**Comment by mergar** [comment]
The license allows forking but usually the project name is changed

**Comment by michael-o** [comment]
I'd like to co-maintain it as well since I use it at work. Created new PRs already. I am `michaelo@` @metalefty @bapt any objections?

**Comment by metalefty** [comment]
Welcome! I will add you later.
